### PR TITLE
Fix compatiblity with Python 3.11+

### PR DESCRIPTION
**Pull request by hroncok**
AttributeError: module 'inspect' has no attribute 'getargspec'

See from https://docs.python.org/3.11/whatsnew/3.11.html#removed

Removed from the inspect module:

> the getargspec function, deprecated since Python 3.0; use inspect.signature() or inspect.getfullargspec() instead.

Downstream bug report: https://bugzilla.redhat.com/show_bug.cgi?id=2095027

### CODE REVIEW

**Review by glensc — APPROVED** [review]

**Review by Conan-Kudo — APPROVED** [review]

**Review by mikebonnet — APPROVED** [review]

**Comment by mikebonnet** [comment]
Can we get this merged?

**Comment by hroncok** [comment]
The latest commit to this repo was in 2020. I don't think it is maintained anymore.

**Review by smooge — APPROVED** [review]
The fix is easy and should work

**Comment by smooge** [comment]
Currently it needed a review from someone with approval.. looks like thats me. Now it is waiting for a required travis-ci which won't happen. I am going to merge

**Comment by smooge** [comment]
And yes, fedmsg is considered pretty dead code from what I can tell.

**Comment by hroncok** [comment]
> The fix is easy and should work

It works. It's been backported in Fedora 37 since the Python upgrade.

**Pull request merged**
